src/jdk.jdi/share/classes/com/sun/tools/jdi/ThreadReferenceImpl.java

```diff
@@ -273,7 +273,18 @@ public void stop(ObjectReference throwable) throws InvalidTypeException {
             JDWP.ThreadReference.Stop.process(vm, this,
                                          (ObjectReferenceImpl)throwable);
         } catch (JDWPException exc) {
-            throw exc.toJDIException();
+            switch (exc.errorCode()) {
+            case JDWP.Error.OPAQUE_FRAME:
+                assert isVirtual(); // can only happen with virtual threads
+                throw new OpaqueFrameException();
```

Comment:
The implementation tends to only include an exception message when it helps to clarify the reason for the exception. The spec only gives one possible reason for this exception, so no clarification should be needed. Plus it would be hard to meaningfully convey this reason in a short message. Here's what the spec says:

     * @throws OpaqueFrameException if the thread is a suspended
     * virtual thread and the implementation was unable to throw an
     * asynchronous exception from this frame

Reply:
Okay, thanks.